### ISSUE TITLE
Does not work in jruby

### ISSUE THREAD

**Issue by HoneyryderChuck** [janko/sequel-activerecord_connection#4]
Right now, JRuby doesn't work out of the box. Although the sequel generation is properly enabled, extending this plugin will fail with:

```
Sequel::ActiveRecordConnection::Error: unsupported adapter: jdbc
 extended at /dev/roda-oauth/.bundle/jruby/2.5.0/bundler/gems/sequel-activerecord_connection-2853a2a8b9ed/lib/sequel/extensions/activerecord_connection.rb:13
extend at org/jruby/RubyKernel.java:2183
register_extension at /dev/roda-oauth/.bundle/jruby/2.5.0/bundler/gems/sequel-5d056c3c8388/lib/sequel/database/misc.rb:59
extension at /dev/roda-oauth/.bundle/jruby/2.5.0/bundler/gems/sequel-5d056c3c8388/lib/sequel/database/misc.rb:211
each at org/jruby/RubyArray.java:1809
extension at /dev/roda-oauth/.bundle/jruby/2.5.0/bundler/gems/sequel-5d056c3c8388/lib/sequel/database/misc.rb:207
<main> at /dev/roda-oauth/test/rails_app/config/initializers/sequel.rb:11
                                
```

I think it has to do with this line here: https://github.com/janko/sequel-activerecord_connection/blob/master/lib/sequel/extensions/activerecord_connection.rb#L10

**Comment by janko** [comment]
Yes, I chose to leave out JRuby support from of the initial release, I should probably mention this in the README.

The reason is that it's generally fairly of difficult to get an adapter working, because it requires basic understanding of the adapter code on both Active Record and Sequel side. And it seemed to me that adding JRuby support would require writing integration for each individual adapter from [activerecord-jdbc-adapter](https://github.com/jruby/activerecord-jdbc-adapter) gem, together with the matching Sequel JDBC adapter, which is a pretty large matrix.

I will probably start at least with `sqlite3`, `postgresql` and `mysql` adapters at some point, and see how that goes. Contributions are more than welcome as well.

**Comment by HoneyryderChuck** [comment]
Didn't think about that, it's actually understandable given how much specific code is built into adapters. 

I think that maybe a first step could be to add a jruby-enabled CI (and maybe don't test all rails versions with it), and allow it to fail.

> it seemed to me that adding JRuby support would require writing integration for each individual adapter from activerecord-jdbc-adapter gem.

Perhaps you're right. However, I had a quick look at what this gem wraps, and if I understood correctly, it basically invokes the sql query on the active record raw connection, while returning a close-enough result back, right? Because (and again, this was a quick look-around) [the activerecord jdbc adapter abstracts the execute method away](https://github.com/jruby/activerecord-jdbc-adapter/blob/1ca0b035f0c1a00c0f669f1ba4675c51db9f402d/lib/arjdbc/jdbc/adapter.rb#L255), so maybe it's just a matter of inserting glue code only for JDBC?

I'll see if I can start with the CI step.

**Comment by janko** [comment]
The `postgresql` integration could pretty much just call `#execute`, but `mysql2` integration required more logic around replicating connection option Sequel expects, while for `sqlite3` I had to skip `#execute` entirely and call the methods on the SQLite object directly because Sequel required specific methods to be called for different type of queries.

I'm hoping the JDBC integration will be easier, thanks for investigating :)

**Comment by HoneyryderChuck** [comment]
I've started having a look at this today, I'll let you know as I progress.

**Comment by HoneyryderChuck** [comment]
I guess this is taken care of, let me know once you release it.

**Comment by janko** [comment]
Released 0.2.4 with the new JRuby support 🎉
